compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1406,10 +1486,20 @@ static LogicalResult setElementwiseGenericOpRootConfig(
   tileSizes.push_back(vecTileSizes);
   tileSizes.push_back(zeros);
 
-  auto passPipeline =
-      genericOp.hasTensorSemantics()
-          ? DispatchLoweringPassPipeline::CPUDoubleTilingPeelingExpert
-          : DispatchLoweringPassPipeline::CPUBufferOpsTileAndVectorize;
+  LLVM_DEBUG(KD_DBGS() << "Final tile sizes for element-wise op: " << tileSizes
+                       << "\n");
+
+  DispatchLoweringPassPipeline passPipeline;
+  if (genericOp.hasBufferSemantics()) {
+    passPipeline = DispatchLoweringPassPipeline::CPUBufferOpsTileAndVectorize;
+  } else {
+    if (vecPreProcStrategy == VectorPreProcStrategy::Peeling) {
+      passPipeline = DispatchLoweringPassPipeline::CPUDoubleTilingPeelingExpert;
+    } else {
+      passPipeline = DispatchLoweringPassPipeline::CPUDoubleTilingExpert;
+    }
+  }
```

Comment:
nit: I think we can save one level of nesting here.

```suggestion
  } else if (vecPreProcStrategy == VectorPreProcStrategy::Peeling) {
      passPipeline = DispatchLoweringPassPipeline::CPUDoubleTilingPeelingExpert;
  } else {
      passPipeline = DispatchLoweringPassPipeline::CPUDoubleTilingExpert;
  }
```